cpp/src/arrow/sparse_tensor.h

```diff
@@ -220,14 +261,66 @@ class ARROW_EXPORT SparseCSRIndex : public internal::SparseIndexBase<SparseCSRIn
       return Status::OK();
     }
 
-    return Status::Invalid("shape length is inconsistent with the CSR index");
+    return Status::Invalid("shape length is inconsistent with the ", ToString());
   }
 
  protected:
   std::shared_ptr<Tensor> indptr_;
   std::shared_ptr<Tensor> indices_;
 };
 
+}  // namespace internal
+
+// ----------------------------------------------------------------------
+// SparseCSRIndex class
+
+/// \brief EXPERIMENTAL: The index data for a CSR sparse matrix
+///
+/// A CSR sparse index manages the location of its non-zero values by two
+/// vectors.
+///
+/// The first vector, called indptr, represents the range of the rows; the i-th
+/// row spans from indptr[i] to indptr[i+1] in the corresponding value vector.
+/// So the length of an indptr vector is the number of rows + 1.
+///
+/// The other vector, called indices, represents the column indices of the
+/// corresponding non-zero values.  So the length of an indices vector is same
+/// as the number of non-zero-values.
+class ARROW_EXPORT SparseCSRIndex
+    : public internal::SparseCSXIndex<SparseCSRIndex,
+                                      internal::SparseMatrixCompressedAxis::ROW> {
+ public:
+  constexpr static char const* TYPE_NAME = "SparseCSRIndex";
```

Comment:
Also, make this `kTypeName`.

Reply:
OK.  I should rename `format_id` to `kFormatId`?

Reply:
Hmm. Our current convention would be to keep `format_id`. I'm not sure about `kTypeName` vs. `type_name`, TBH. But it definitely shouldn't be all-uppercase `TYPE_NAME`.
@wesm thoughts?